scripts/quests/bastok/Chips.lua

```diff
@@ -0,0 +1,122 @@
+-----------------------------------
+-- Chips
+-----------------------------------
+-- Log ID: 1, Quest ID: 82
+-----------------------------------
+local metalworksID = zones[xi.zone.METALWORKS]
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.BASTOK, xi.quest.id.bastok.CHIPS)
+
+quest.reward =
+{
+    item = xi.item.CCB_POLYMER_PUMP,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            (player:getCurrentMission(xi.mission.log_id.COP) > xi.mission.id.cop.ONE_TO_BE_FEARED or
+            (player:getCurrentMission(xi.mission.log_id.COP) == xi.mission.id.cop.ONE_TO_BE_FEARED and
+            player:getCharVar('Mission[6][638]Status') >= 1)) and
+            not quest:getMustZone(player)
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Ghebi_Damomohe'] = quest:progressEvent(169),
+
+            onEventFinish =
+            {
+                [169] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                    else quest:setMustZone(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Cid'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, {
+                        xi.item.CARMINE_CHIP,
+                        xi.item.CYAN_CHIP,
+                        xi.item.GRAY_CHIP
+                        }) and
+                        not player:hasItem(xi.item.CCB_POLYMER_PUMP)
+                    then
+                        return quest:progressEvent(883)
+                    elseif
```

Comment:
Is the player really limited to only one pump? Its ex, not rare

Reply:
![image](https://github.com/LandSandBoat/server/assets/13935086/5c2f8127-b3d7-44fd-a3f4-6d91e56fcfdd)
It appears from capture, that is the case.

Reply:
Yea I thought the same thing had to double check the capture at first - anyways thanks for the feedback. Made alterations. <3